src/components/Animation/AnimationRectangle.vue

```diff
@@ -0,0 +1,292 @@
+            let layerTitle = `• ${this.$t(visibleLayers[i].get("layerName"))}`;
+            let fontSize = baseFont;
+            ctx.font = fontSize + "px sans-serif";
+            metrics = ctx.measureText(layerTitle);
+            while (metrics.width > mapWidth - 320 && fontSize > minFont) {
+              fontSize -= 1;
+              ctx.font = fontSize + "px sans-serif";
+              metrics = ctx.measureText(layerTitle);
+            }
+            ctx_h += baseFont + 8;
+          }
+          // Must be divisible by 2 otherwise encoder.initialize() will fail
+          ctx_h = 2 * Math.ceil(ctx_h / 2);
+          if (ctx_h < 40) ctx_h = 40;
+        }
+        let animetOffset = 0;
+        if (!isLayerListShown) {
+        } else if (
+          !this.$mapCanvas.mapObj.getLayers().getArray()[0].get("visible")
+        ) {
+          if (mapWidth < 1080) {
+            animetOffset = 21;
+          } else {
+            animetOffset = 23;
+          }
+        } else {
+          if (mapWidth < 1080) {
+            animetOffset = 30;
+          } else {
+            animetOffset = 34;
+          }
+        }
+
+        let animetAttr = this.$t("MadeWithAniMet");
+        if (mapWidth < 1080) {
+          canvasTxt.fontSize = 13;
+        } else {
+          canvasTxt.fontSize = 15;
+        }
+        canvasTxt.align = "left";
+        ctx.font = canvasTxt.fontSize + "px sans-serif";
+        metrics = ctx.measureText(animetAttr);
+
+        if (numModelRuns !== 0 && isLayerListShown) {
+          let minHeight;
+          if (numModelRuns === 1) {
+            minHeight = baseFont + animetOffset;
+          } else {
+            const lastMRIndex = modelRuns.findLastIndex((mr) => mr !== "");
+            minHeight = (lastMRIndex + 1) * (baseFont + 8) + animetOffset - 6;
+          }
+          ctx_h = ctx_h >= minHeight ? ctx_h : minHeight;
+        } else if (!isLayerListShown && numModelRuns === 0) {
+          if (
+            !this.$mapCanvas.mapObj.getLayers().getArray()[0].get("visible")
+          ) {
+            ctx_h = 24;
+          }
+          ctx_w = metrics.width + 12;
+        } else if (
+          (numModelRuns === 0 && isLayerListShown) ||
+          (numModelRuns !== 0 && !isLayerListShown)
+        ) {
+          if (
+            !this.$mapCanvas.mapObj.getLayers().getArray()[0].get("visible")
+          ) {
+            ctx_h = 24;
+          }
+        }
+        let outputHeaderCanvas = 50;
+        if (mapWidth < 1080) {
+          outputHeaderCanvas = 30;
+        }
+        this.setHeaderFooterPreviewHeight(
+          outputHeaderCanvas,
+          ctx_w,
+          ctx_h,
+          mapWidth
+        );
+      }
+    },
+    getModelRuns() {
+      let modelRuns = [];
+      let visibleLayers = this.$mapLayers.arr
+        .filter((l) => {
+          return l.get("layerVisibilityOn");
+        })
+        .reverse();
+      for (let i = 0; i < visibleLayers.length; i++) {
```

Comment:
Likewise, change to use a `const` variable to avoid having to recalculate `.length` each loop.